Website/Components/TagCloudCore.cs

```diff
@@ -28,8 +30,30 @@ public class TagCloudCore : ITagCloudCore
             set { _tags = value; }
         }
 
-        public TagCloudCore(IBlogManager blogManagerInstance)
+        private string[] m_SortingNames = null;
+
+        public string[] GetSortNames(string sortingOptions)
         {
+                if (m_SortingNames != null)
+                {
+                    return m_SortingNames;
+                }
+                if (string.IsNullOrEmpty(sortingOptions))
+                {
+                    return new string[0];
+                }
+                var database = Context.Database;
```

Comment:
Avoid use of Sitecore.Context as it makes testing and reuse difficult. Instead, we've already got the CurrentBlog property, use the Database off that.